tests/test_ace.rb

```diff
@@ -0,0 +1,134 @@
+# Copyright (c) 2014-2015 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require_relative 'ciscotest'
+require_relative '../lib/cisco_node_utils/acl'
+require_relative '../lib/cisco_node_utils/ace'
+
+# TestX__CLASS_NAME__X - Minitest for X__CLASS_NAME__X node utility class
+class TestAceV4 < CiscoTestCase
+  attr_reader :acl_name, :seqno, :afi
+
+  def setup
+    # setup runs at the beginning of each test
+    super
+    @acl_name = 'test-foo-1'
+    @seqno = 10
+    @afi = 'ip'
+    no_ip_access_list_foo
+  end
+
+  def teardown
+    # teardown runs at the end of each test
+    no_ip_access_list_foo
+    super
+  end
+
+  def no_ip_access_list_foo
+    # Turn the feature off for a clean test.
+    config('no ip access-list ' + @acl_name)
+  end
+
+  # TESTS
+
+  def test_create_destroy_ipv4_ace_one
```

Comment:
You're going to have a lot of duplicated code if you create separate v4 and v6 tests that are essentially testing the same things. You can structure your code to use a common test method that's used for both - see the ACL test code here for an example: https://github.com/cisco/cisco-network-node-utils/pull/111/files#diff-5dde18e7fed82ed58d46532b7f59c873R43

The only difference is that you'll need to pass in some v4/v6 specific variables as well, which you could do by setting up a hash first, e.g.:

```
  def test_create_destroy_ace_one
    props = {
      'ip' => {
        :afi => 'ip'
        :acl_name => @acl_name_v4,
        :proto' => 'tcp',
        :src_addr => '7.8.9.6 2.3.4.5',
        :src_port => 'eq 40',
        :dst_addr => '1.2.3.4/32',
        :dst_port => 'neq 20',
        :option => 'precedence critical',
      },
      'ipv6' => {
        :afi => 'ipv6'
        :acl_name => @acl_name_v6,
        :proto => '6',
        :src_addr => 'addrgroup fi',
        :src_port => 'eq 30',
        :dst_addr => '1::7/32',
        :dst_port => 'neq 20',
        :option => 'dscp cs2 fragments packet-length eq 30',
      }
    }

    %w(ip ipv6).each do |afi|
      create_destroy_ace_one(props[afi])
    end
  end

  def create_destroy_ace_one(p)

    # Test seqno 10 with permit
    p[:action] = 'permit'
    ace = Ace.new(p[:afi], p[:acl_name], 10)
    ace.ace_set(p)
    assert_equal(p[:action], ace.action)
    assert_equal(p[:src_addr], ace.src_addr)
    assert_equal(p[:src_port], ace.src_port)
      ...
    # I don't think we need assert_show_match for this test. May revisit.

    # Test seqno 20 with deny any
    p[:action] = 'deny'
    p[:src_addr] = 'any'
    p[:dst_addr] = 'any'
    ace = Ace.new(p[:afi], p[:acl_name], 20)
    ace.ace_set(p)
    assert_equal(p[:action], ace.action)
    assert_equal(p[:src_addr], ace.src_addr)
    assert_equal(p[:src_port], ace.src_port)
      ...
```